Sources/TuistKit/Linter/TargetLinter.swift

```diff
@@ -40,6 +41,24 @@ class TargetLinter: TargetLinting {
     }
 
     // MARK: - Fileprivate
+    
+    /// Verifies that the bundle identifier doesn't include characters that are not supported.
+    ///
+    /// - Parameter target: Target whose bundle identified will be linted.
+    /// - Returns: An array with a linting issue if the bundle identifier contains invalid characters.
+    fileprivate func lintBundleIdentifier(target: Target) -> [LintingIssue] {
+        let bundleIdentifier = target.bundleId
+        
+         var allowed = CharacterSet.alphanumerics
```

Comment:
`alphanumerics` in swift contain Unicode General Categories & Code Points for L*, M* & N*. This means it will contain characters which are not allowed in a bundle identifier. See a complete list here:

https://www.compart.com/en/unicode/category

For example:

```swift
CharacterSet.alphanumerics.contains("\u{0301}") // true
```

where U+0301 (◌́) is this nonspacing mark and a valid alphanumeric, but invalid character for bundle identifier. 
I think we need to be a little more direct in usage and only allow `A-Z`,`a-z`,`0-9`, hyphen (`-`), and period (`.`) characters.

Reply:
The wording from the original error message is slightly misleading. It means the only the alphanumeric character subset which it describes. (`A-Z`,`a-z`,`0-9`, `-` and `.`)